XiEditor/EditView.swift

```diff
@@ -464,6 +466,7 @@ class EditView: NSView, NSTextInputClient, TextPlaneDelegate {
                 lineCache.setAssoc(lineIx, assoc: assoc)
                 textLines.append(textLine)
             }
+            maxLineWidth = max(maxLineWidth, textLine.width)
```

Comment:
I'm slightly uncomfortable with drawing being side-effecting. I'd prefer if the width recalculation were done separately, triggered by updates. It occurs to me that logic will need full access to the TextLine objects, which means that we'd have to move the "get cached TextLine" logic into a separate method. Among other things, I'm pretty sure this causes a double draw when the width increases.

As a side note, side effects from drawing are also used by the IME logic.

Reply:
I tried an approach similar to what you'd described, but because LineCache has a generic payload I can't even really access the `TextLine`s. (It occurred to me just now that I could write an extension for `LineCache where T == TextLine`, which might work). 

Also, when text lines arrive we tend to be blocked for drawing; we certainly don't want to introduce additional blocking gets earlier. So doing the calculation while drawing felt like an expedient compromise.

Ideally, we would have width measurement, and core would do a bunch of the work here for us; I  think that should be the final goal.

In terms of redundant draws with the current implementation, they will be happening but only in the case where a draw introduces a new 'longest line', so the amortized cost shouldn't be too bad.